v2/api/apimanagement/v1api20220801/structure.txt

```diff
@@ -0,0 +1,571 @@
+// Code generated by azure-service-operator-codegen. DO NOT EDIT.
+github.com/Azure/azure-service-operator/v2/api/apimanagement/v1api20220801
+├── APIVersion: Enum (1 value)
+│   └── "2022-08-01"
+├── Service: Resource
+│   ├── Owner: github.com/Azure/azure-service-operator/v2/api/resources/v1apiv20191001.ResourceGroup
+│   ├── Spec: Object (24 properties)
+│   │   ├── AdditionalLocations: Object (7 properties)[]
+│   │   │   ├── DisableGateway: *bool
+│   │   │   ├── Location: *string
+│   │   │   ├── NatGatewayState: *Enum (2 values)
+│   │   │   │   ├── "Disabled"
+│   │   │   │   └── "Enabled"
+│   │   │   ├── PublicIpAddressId: *string
+│   │   │   ├── Sku: *Object (2 properties)
+│   │   │   │   ├── Capacity: *int
+│   │   │   │   └── Name: *Enum (6 values)
+│   │   │   │       ├── "Basic"
+│   │   │   │       ├── "Consumption"
+│   │   │   │       ├── "Developer"
+│   │   │   │       ├── "Isolated"
+│   │   │   │       ├── "Premium"
+│   │   │   │       └── "Standard"
+│   │   │   ├── VirtualNetworkConfiguration: *Object (1 property)
+│   │   │   │   └── SubnetResourceReference: *genruntime.ResourceReference
+│   │   │   └── Zones: string[]
+│   │   ├── ApiVersionConstraint: *Object (1 property)
+│   │   │   └── MinApiVersion: *string
+│   │   ├── AzureName: Validated<string> (3 rules)
+│   │   │   ├── Rule 0: MaxLength: 50
+│   │   │   ├── Rule 1: MinLength: 1
+│   │   │   └── Rule 2: Pattern: "^[a-zA-Z](?:[a-zA-Z0-9-]*[a-zA-Z0-9])?$"
+│   │   ├── Certificates: Object (4 properties)[]
+│   │   │   ├── Certificate: *Object (3 properties)
+│   │   │   │   ├── Expiry: *string
+│   │   │   │   ├── Subject: *string
+│   │   │   │   └── Thumbprint: *string
+│   │   │   ├── CertificatePassword: *string
+│   │   │   ├── EncodedCertificate: *string
+│   │   │   └── StoreName: *Enum (2 values)
+│   │   │       ├── "CertificateAuthority"
+│   │   │       └── "Root"
+│   │   ├── CustomProperties: map[string]string
+│   │   ├── DisableGateway: *bool
+│   │   ├── EnableClientCertificate: *bool
+│   │   ├── HostnameConfigurations: Object (11 properties)[]
+│   │   │   ├── Certificate: *Object (3 properties)
+│   │   │   │   ├── Expiry: *string
+│   │   │   │   ├── Subject: *string
+│   │   │   │   └── Thumbprint: *string
+│   │   │   ├── CertificatePassword: *string
+│   │   │   ├── CertificateSource: *Enum (4 values)
+│   │   │   │   ├── "BuiltIn"
+│   │   │   │   ├── "Custom"
+│   │   │   │   ├── "KeyVault"
+│   │   │   │   └── "Managed"
+│   │   │   ├── CertificateStatus: *Enum (3 values)
+│   │   │   │   ├── "Completed"
+│   │   │   │   ├── "Failed"
+│   │   │   │   └── "InProgress"
+│   │   │   ├── DefaultSslBinding: *bool
+│   │   │   ├── EncodedCertificate: *string
+│   │   │   ├── HostName: *string
+│   │   │   ├── IdentityClientId: *string
```

Comment:
We probably want to be able to take this from configMap. Add:
```
      HostnameConfiguration:
        IdentityClientId:
          $importConfigMapMode: optional
```

Reply:
Done